src/sourceMachine.ts

```diff
@@ -171,13 +169,14 @@ export const makeSourceMachine = (params: {
 
   return createMachine<typeof sourceModel>(
     {
-      initial: 'checking_if_on_legacy_url',
+      initial: 'checking_initial_data',
```

Comment:
this seemed like a priority condition - if we have data then we definitely don't need to check anything else